plugin/evm/atomic/vm/syncervm_test.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/ava-labs/coreth/plugin/evm/atomic"
+	"github.com/ava-labs/coreth/plugin/evm/atomic/atomictest"
+	"github.com/ava-labs/coreth/plugin/evm/extension"
+	"github.com/ava-labs/coreth/plugin/evm/vmtest"
+	"github.com/ava-labs/coreth/predicate"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/types"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAtomicSyncerVM(t *testing.T) {
+	importAmount := 2000000 * units.Avax // 2M avax
+	for _, test := range vmtest.SyncerVMTests {
+		includedAtomicTxs := make([]*atomic.Tx, 0)
+
+		t.Run(test.Name, func(t *testing.T) {
+			genFn := func(i int, vm extension.InnerVM, gen *core.BlockGen) {
+				atomicVM, ok := vm.(*VM)
+				require.True(t, ok)
+				b, err := predicate.NewResults().Bytes()
+				if err != nil {
+					t.Fatal(err)
+				}
+				gen.AppendExtra(b)
+				switch i {
+				case 0:
+					// spend the UTXOs from shared memory
+					importTx, err := atomicVM.newImportTx(atomicVM.Ctx.XChainID, vmtest.TestEthAddrs[0], vmtest.InitialBaseFee, vmtest.TestKeys[0:1])
+					require.NoError(t, err)
+					require.NoError(t, atomicVM.AtomicMempool.AddLocalTx(importTx))
+					includedAtomicTxs = append(includedAtomicTxs, importTx)
+				case 1:
+					// export some of the imported UTXOs to test exportTx is properly synced
+					state, err := vm.Ethereum().BlockChain().State()
+					if err != nil {
+						t.Fatal(err)
+					}
+					wrappedStateDB := extstate.New(state)
+					exportTx, err := atomic.NewExportTx(
+						atomicVM.Ctx,
+						atomicVM.CurrentRules(),
+						wrappedStateDB,
+						atomicVM.Ctx.AVAXAssetID,
+						importAmount/2,
+						atomicVM.Ctx.XChainID,
+						vmtest.TestShortIDAddrs[0],
+						vmtest.InitialBaseFee,
+						vmtest.TestKeys[0:1],
+					)
+					require.NoError(t, err)
+					require.NoError(t, atomicVM.AtomicMempool.AddLocalTx(exportTx))
+					includedAtomicTxs = append(includedAtomicTxs, exportTx)
+				default: // Generate simple transfer transactions.
+					pk := vmtest.TestKeys[0].ToECDSA()
+					tx := types.NewTransaction(gen.TxNonce(vmtest.TestEthAddrs[0]), vmtest.TestEthAddrs[1], common.Big1, params.TxGas, vmtest.InitialBaseFee, nil)
+					signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.Ethereum().BlockChain().Config().ChainID), pk)
+					require.NoError(t, err)
+					gen.AddTx(signedTx)
+				}
+			}
+			newVMFn := func() (extension.InnerVM, dummy.ConsensusCallbacks) {
+				vm := newAtomicTestVM()
+				return vm, vm.createConsensusCallbacks()
+			}
+
+			afterInit := func(t *testing.T, params vmtest.SyncTestParams, vmSetup vmtest.SyncVMSetup, isServer bool) {
+				atomicVM, ok := vmSetup.VM.(*VM)
+				require.True(t, ok)
+
+				alloc := map[ids.ShortID]uint64{
+					vmtest.TestShortIDAddrs[0]: importAmount,
+				}
+
+				for addr, avaxAmount := range alloc {
+					txID, err := ids.ToID(hashing.ComputeHash256(addr.Bytes()))
+					if err != nil {
+						t.Fatalf("Failed to generate txID from addr: %s", err)
```

Comment:
Could swap `t.Fatal` for `require.NoError(t, err, "Failed to generate txID from addr")`, though I do remember that the linter sometimes preferred `t.Fatal` for some strange reason, so take this with a grain of salt.